### DISCUSSION TITLE
Models?

### DISCUSSION THREAD

**Discussion by Kallamamran** [TheCompAce/Auto-GPT-Powershell#6] [General]
Will this be able to run models like GPT4-X-Alpaca, Pygmalion and Vicuna?

**Comment by TheCompAce** [comment]
I hope to expand it, but it seems the offline ones are either not up to snuff yet, or just don't run without a massive amount of RAM and PC power.

**Reply by Kallamamran** [reply]
Isn't "gpt4all-lora-quantized.bin" a local model? 🤔

**Reply by TheCompAce** [reply]
yes I read it with the exe file.

**Reply by TheCompAce** [reply]
you dont need the scripts, you can run the exe with "-h" to get the arguments to just run the model.

**Reply by Kallamamran** [reply]
So what does the scripts do then 😊

**Reply by TheCompAce** [reply]
It just loops from a start prompt, mutates it using plugins and sends it to the GPT, then it takes the response and mutates it, saves it to the prompt, and does it again, again, etc.

**Reply by Kallamamran** [reply]
Oh ok 👍🏻

**Comment by TheCompAce** [comment]
I looked into adding "StableLM" as a option, but my PC can not handle it. I will plan on adding the script files I used to test it into a StableLM folder if someone else wants to try it. I plane on making a Option to use it, because why not, but I will not be able to test it, after that.

**Reply by Kallamamran** [reply]
Don't know what that is, but since I haven't been able to successfully en it yet I'm probably not the best test driver

**Comment by TheCompAce** [comment]
Here is a video about StableLM
https://www.youtube.com/watch?v=qXtJjoEfTnA

It almost killed my PC before it ran out of Memory, I made a script that cuts down some settings, but it was not enough.
As for testing this, the offline does not work, well it does just the AI is stupid, so I was hopping this was going to fix that problem.  But I am working on a new version, to give access to more advance options, but also added a Easy setup option for OpenAI.

**Comment by Kallamamran** [comment]
Sounds very interesting! I can run StableLM if you want to test something. I'm on 24GB VRAM

**Reply by TheCompAce** [reply]
Hopefully I will have the add-ons (and fixes) wrapped up today for the next update, and it should have StableLM included (I can test the interface, but my 10VRAM, can only run the smaller model, so it just doesn't work for me.).  I have added "BARK" (for Text To Speech) and "Stable Diffusion" (for a offline solution for DALLE), I also added serval menu options to easily setup the different sections.  I am also looking into adding "AudioLMD" (Text to Sound Effects, but seems to be missing a DLL, but I spent all of 30 mins. looking into it before bed.) 

So as of right now I have plugins for Text To Image, Text to Speech, and the main GPT connections.  Hopefully I can get back started on the "Code Saver" plugin that will implement all this to create a program or game, just by supplying a simple sentence or two, and it will make the code/images/wave files/etc.
